### ISSUE TITLE
Network viz: When nodes clicked, their labels should show regardless of if the label is visible

### ISSUE THREAD

**Issue by asizemore** [VEuPathDB/web-monorepo#1127]
In the network viz, we added the ability to remove node labels from normal view. But, if we click on the nodes, we should override this hidden-label feature and show the node label so that the node label of the highlighted (clicked) node is visible. Clicking the node would not turn on the visibility of the label if it was off, and if the user unclicks a node that node's label should disappear if it's not supposed to be shown (hopefully that all made sense!). 

Screenshot of current behavior:
https://github.com/VEuPathDB/web-monorepo/assets/11710234/af03f693-357c-4f88-aeee-a313c3207804

**Comment by bobular** [comment]
Also in the select/search box, the "All 9 selected" should be "All 9 labels shown" (Danica's suggestion)

**Comment by asizemore** [comment]
Sorry @moontrip the description is confusing 😄 ! Let me try another way:
Right now the label is shown if that node is listed in the Visible Node Labels box, or if the cursor is hovering over the node. However, if we click on a node that is not listed in the VIsible Node Labels box, then its label is not shown. This last part is the part we should flip. So if a node is clicked, the label is shown in the plot, just like on hover. But, (this is the part that gets confusing i think), clicking the node and thus showing the label should _not_ add that node to the Visible Node Labels box if it wasn't already there. 

In other words, we should always show the node label when we click on a node. Does that help to clarify? Let me know if there's anything else that doesn't make sense!

**Comment by moontrip** [comment]
@asizemore Thank you for your information. 👍 Now I understand the meaning but I am not quite sure if it is doable without handling the node control (e.g., adding/removing the node label to the list of visible node labels). Is there any specific reason why it should not be added to the Visible Node Labels box? 

Another question is the last sentence in the ticket's description, citing "if the user unclicks a node that node's label should disappear if it's not supposed to be shown" Does it mean "if a user clicks the highlighted node again, it should hide the node label" ?  (assuming that highlighted node already showed its node label due to the clicking node action)

I still think that those behaviors are better to be related to Node Lable Control so that UI/UX become consistent for user perspective and it is more doable for implementation aspect (😄). Though I will try your intention at first.

**Comment by moontrip** [comment]
@asizemore I think I made the click event as you described: I was overthinking and spent time too much with complicated logics but I realized that there was a simpler solution later. Please take a look at https://github.com/VEuPathDB/web-monorepo/pull/1130. As far as I know backend servers are not working so I just tested with storybook. I guess it would work with network Viz but will confirm it when the server is ready.

p.s. I also tested the approach I mentioned, i.e., using node control (e.g., adding/removing clicked node label), just in case and I think I found the solution for that too, though it seems that you prefer your suggestions 😅
